tests/integration/cloudformation/api/test_stacks.py

```diff
@@ -293,3 +294,160 @@ def stack_process_is_finished(cfn_client, stack_name):
+    )
+
+    drift_detection = cfn_client.detect_stack_resource_drift(
+        StackName=stack.stack_name, LogicalResourceId="Function76856677"
+    )
+
+    snapshot.match("drift_detection", drift_detection)
+
+
+@pytest.mark.aws_validate
+@pytest.mark.skip(reason="Not Implemented")
+def test_linting_error_during_creation(cfn_client):
+    stack_name = f"stack-{short_uid()}"
+    bad_template = {"Resources": "", "Outputs": ""}
+
+    with pytest.raises(botocore.exceptions.ClientError) as ex:
+        cfn_client.create_stack(StackName=stack_name, TemplateBody=json.dumps(bad_template))
+
+    error_response = ex.value.response
+
+    assert error_response["Error"]["Code"] == "ValidationError"
+    assert "Template format error" in error_response["Error"]["Message"]
+
+
+@pytest.mark.aws_validated
+@pytest.mark.skip(reason="feature not implemented")
+def test_notifications(
+    sqs_client,
+    cfn_client,
+    deploy_cfn_template,
+    sns_create_topic,
+    is_stack_created,
+    is_stack_updated,
+    sqs_create_queue,
+    sns_create_sqs_subscription,
+):
+    stack_name = f"stack-{short_uid()}"
+    topic_arn = sns_create_topic()["TopicArn"]
+    sqs_url = sqs_create_queue()
+    sns_create_sqs_subscription(topic_arn, sqs_url)
+
+    template = load_file(
+        os.path.join(os.path.dirname(__file__), "../../templates/sns_topic_parameter.yml")
+    )
+    cfn_client.create_stack(
+        StackName=stack_name,
+        NotificationARNs=[topic_arn],
+        TemplateBody=template,
+        Parameters=[{"ParameterKey": "TopicName", "ParameterValue": f"topic-{short_uid()}"}],
+    )
+
+    assert wait_until(is_stack_created(stack_name))
+
+    template = load_file(
+        os.path.join(os.path.dirname(__file__), "../../templates/sns_topic_parameter.yml")
+    )
+    cfn_client.update_stack(
+        StackName=stack_name,
+        TemplateBody=template,
+        Parameters=[
+            {"ParameterKey": "TopicName", "ParameterValue": f"topic-{short_uid()}"},
+        ],
+    )
+    assert wait_until(is_stack_updated(stack_name))
+
+    messages = {}
+
+    def _assert_messages():
+        sqs_messages = sqs_client.receive_message(QueueUrl=sqs_url)["Messages"]
+        for sqs_message in sqs_messages:
+            sns_message = json.loads(sqs_message["Body"])
+            messages.update({sns_message["MessageId"]: sns_message})
+
+        # Assert notifications of resources created
+        assert [message for message in messages.values() if "CREATE_" in message["Message"]]
+
+        # Assert notifications of resources deleted
+        assert [message for message in messages.values() if "UPDATE_" in message["Message"]]
+
+        # Assert notifications of resources deleted
+        assert [message for message in messages.values() if "DELETE_" in message["Message"]]
+
+    try:
+        retry(_assert_messages, retries=10, sleep=2)
+    finally:
+        cfn_client.delete_stack(StackName=stack_name)
```

Comment:
We have a `cleanups` fixture that is better suited for this purpose.

It's just a list that you can fill with callbacks to be executed in reverse order after the test finishes.

The general pattern looks like this:

```python
def test_something(cleanups):
    result = client.create_something(...)
    cleanups.append(lambda: client.delete_something(Id=result['id'])

    ... rest of test
```